### ISSUE TITLE
Feature requests - sort on redstone trigger and when full

### ISSUE THREAD

**Issue by jw-taylor** [flokol120/Spigot-Item-Chest-Sorter#26]
Hi, I'm running this on my server and it's being used beneath an iron farm - golem dies, iron and poppies go through hoppers into a chest and the chest sorts them separately.

Because the items come in through hoppers, the plugin sorts the items each time one goes into the chest, so it's running a lot, using a bunch of server resources.

Would it be possible to trigger item sorting from the chest when it receives a redstone signal, please? That way we could have a comparator next to a chest to detect when it is full, output a redstone signal back to the chest, and it sorts itself (so it only runs once every few hours instead of multiple times a second).

Also, could you add the option please for the chest to sort itself when it is full? That would mean we don't even need redstone.

Thanks!

**Comment by flokol120** [comment]
#9

**Comment by flokol120** [comment]
@jw-taylor I like the redstone idea but can't make any promises if I can implement it. New terrain for me :P

What do you mean by "sort itself when full"? Items should be stacked on top of each other :thinking:

**Comment by jw-taylor** [comment]
> What do you mean by "sort itself when full"? Items should be stacked on top of each other

Sorry I was unclear. What I mean is, in the same way the chest "sends" the items when you close it, and could "send" the items when it receives a redstone trigger, could it also "send" the items when it gets full? So a hopper puts iron into it, and when it is completely filled instead of me having to go open and close it for it to "send" that iron to a receiver chest, it does it itself?

By "full" I mean when all slots have items in, rather than when all slots have completely full stacks. Because for example if you have a chest that you dump loads of random stuff in (e.g. after coming back from mining or a nether adventure) and throw those into a hopper that fills the chest, it could be that all the slots are full but not with full stacks (one slot has 12 cobble in, another has 42 netherrack) and so the hopper gets filled up with items it can't push into the chest.

**Comment by flokol120** [comment]
@jw-taylor Sorting the content of a chest is not what I want to do with this plugin as there are most certainly plugins out there doing that already. But sending contents of a chest on a redstone signal what be do-able I guess.
What that be something you could use, without the sorting feature?

**Comment by jw-taylor** [comment]
Sorry yes, that is what I meant by "sort" - sending the items from the
sender chest to the receivers chests

**Comment by flokol120** [comment]
Please verify if https://github.com/flokol120/Spigot-Item-Chest-Sorter/releases/tag/1.11.0 has your intended feature in it :P

**Comment by flokol120** [comment]
@jw-taylor Do you have time to test it?

**Comment by flokol120** [comment]
closed due to inactivity

**Comment by jw-taylor** [comment]
Hi, I just wanted to apologise profusely for never replying to this. I'm not active on github and I've only just found out today that Gmail has always been sending my github notification emails to my bin - meaning I never knew that you'd implemented this feature.

I feel awful that you implemented a feature I requested and it looks like I just completely blanked you. I'm really, really sorry, and I'm really grateful that you put the effort in to implement it.
